dataprep/clean/clean_duplication_utils.py

```diff
@@ -70,20 +78,22 @@ def _key_collision_cluster(self, cluster_method: str) -> None:
         }
         key_func = key_funcs[cluster_method]
         col = self._col
-        # get the count of each item in the dataframe and remove duplicates
-        counts = self._df[col].value_counts(sort=False)
-        no_dups = self._df.drop_duplicates(subset=[col]).reset_index()
-        # create a column "vals" containing tuples of the form (item, count)
-        no_dups.loc[:, "vals"] = no_dups[col].map(lambda val: (val, counts.loc[val]))
+        df = self._df.drop_duplicates(subset=[col])
         # create a column "key" containing keys created by the given key collision method
-        no_dups.loc[:, "key"] = no_dups[col].map(key_func)
+        df["key"] = df[col].map(key_func)
         # put items with the same key into the same list
-        clusters = no_dups.groupby("key")["vals"].agg(list)
+        agg_list = dd.Aggregation(
+            name="agg_list",
+            chunk=lambda s: s.apply(list),
+            agg=lambda s: s.apply(lambda chunks: list(chain.from_iterable(chunks))),
+        )
+        clusters = df.groupby("key")[col].agg(agg_list)
         clusters = clusters.loc[clusters.map(len) > 1]
+        clusters = clusters.compute()
```

Comment:
With dask, my understanding is that having one `.compute()` is better than multiple; Dask can schedule the computations and perhaps the data can be brought from memory into the caches/registers close to the CPUs once rather than multiple times. If that's correct, I'm wondering if we could try having only one `.compute()` per re-cluster. E.g., something like
``` python
clusters, self._counts = dask.compute(clusters, self._df[self._col].value_counts(sort=False))
```
I think your code would need to be restructured a little to if you think this approach is better